app/controllers/concerns/course/discussion/posts_concern.rb

```diff
@@ -1,21 +1,29 @@
-class Course::Discussion::PostsController < Course::ComponentController
-  before_action :set_topic
-  authorize_resource :topic
-  load_and_authorize_resource :post, through: :topic, class: Course::Discussion::Post.name
+module Course::Discussion::PostsConcern
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_topic
+    authorize_resource :discussion_topic
+    load_and_authorize_resource :post, through: :discussion_topic,
```

Comment:
Are you providing any views? If so, the view path needs to have the default partials and views path in it.